nwb_conversion_tools/datainterfaces/ecephys/spikeglx/spikeglxdatainterface.py

```diff
@@ -1,59 +1,120 @@
+import spikeinterface.extractors as si
 from pynwb.ecephys import ElectricalSeries
 
-from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ..baselfpextractorinterface import BaseLFPExtractorInterface
-from ....utils import get_schema_from_method_signature, get_schema_from_hdmf_class, FilePathType
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
+from ....utils.json_schema import (
+    get_schema_from_method_signature,
+    get_schema_from_hdmf_class,
+    FilePathType,
+    dict_deep_update,
+)
+from ....tools.spikeinterface.spikeinterface import set_recording_channel_property
+
 
+RECORDING_TYPE = Union[si.SpikeGLXRecordingExtractor, se.SpikeGLXRecordingExtractor]
 
-def fetch_spikeglx_metadata(file_path: FilePathType, recording: RecordingExtractor, metadata: dict):
-    file_path = Path(file_path)
-    session_id = file_path.parent.stem
 
-    if isinstance(recording, SubRecordingExtractor):
-        n_shanks = int(recording._parent_recording._meta.get("snsShankMap", [1, 1])[1])
-        session_start_time = datetime.fromisoformat(recording._parent_recording._meta["fileCreateTime"]).astimezone()
+def fetch_spikeglx_metadata(source_path: FilePathType, recording: RECORDING_TYPE, metadata: dict):
+    source_path = Path(source_path)
+    if source_path.is_file():
+        session_id = source_path.parent.stem
     else:
-        n_shanks = int(recording._meta.get("snsShankMap", [1, 1])[1])
-        session_start_time = datetime.fromisoformat(recording._meta["fileCreateTime"]).astimezone()
-    if n_shanks > 1:
-        raise NotImplementedError("SpikeGLX metadata for more than a single shank is not yet supported.")
-    metadata["NWBFile"] = dict(session_start_time=session_start_time.strftime("%Y-%m-%dT%H:%M:%S"))
+        session_id = source_path.stem
+
+    metadata_update = dict(
+        NWBFile=dict(
+            session_id=session_id,
+        ),
+        Ecephys=dict(
+            Electrodes=[
+                dict(name="shank_electrode_number", description="0-indexed channel within a shank."),
+                dict(
+                    name="shank_group_name",
+                    description="The name of the ElectrodeGroup this electrode is a part of.",
+                ),
+            ]
+        ),
+    )
+
+    if isinstance(recording, se.SpikeGLXRecordingExtractor):
+        if isinstance(recording, se.SubRecordingExtractor):
+            current_recording = recording._parent_recording
+        else:
+            current_recording = recording
+        n_shanks = int(current_recording._meta.get("snsShankMap", [1, 1])[1])
+        if n_shanks > 1:
+            raise NotImplementedError("SpikeGLX metadata for more than a single shank is not yet supported.")
+        session_start_time = datetime.fromisoformat(current_recording._meta["fileCreateTime"]).astimezone()
+        metadata_update["NWBFile"]["session_start_time"] = (session_start_time.strftime("%Y-%m-%dT%H:%M:%S"),)
```

Comment:
why is this a tuple?

Reply:
why are we converting from iso format to datetime and then back to iso format?

Reply:
its to get the timezone content. 
```python
dt = datetime.fromisoformat("2020-11-03T10:35:10").astimezone()
datetime.isoformat(dt) # "2020-11-03T10:35:10+5:00"
```